Projects/Cyber-Range-K8/K8_cluster_creation.md

```diff
@@ -0,0 +1,114 @@
+
+# Kubernetes Cluster Creation Notes
+
+## TOC
+- [Kubernetes Cluster Creation Notes](#kubernetes-cluster-creation-notes)
+  - [TOC](#toc)
+  - [Prerequisites](#prerequisites)
+  - [Steps](#steps)
+    - [Install Docker on Each Machine](#install-docker-on-each-machine)
+    - [Install Kubernetes Components on Each Machine](#install-kubernetes-components-on-each-machine)
+  - [Step 1: Initialize the Control Plane](#step-1-initialize-the-control-plane)
+    - [Step 1.1: Set Up kubeconfig for kubectl on Control Plane](#step-11-set-up-kubeconfig-for-kubectl-on-control-plane)
+    - [Step 1.2: Deploy a Pod Network(In This Case, Calico)](#step-12-deploy-a-pod-networkin-this-case-calico)
+      - [Why?](#why)
+  - [Step 2: Join Worker Nodes to the Cluster](#step-2-join-worker-nodes-to-the-cluster)
+  - [Step 3: Verify the Cluster](#step-3-verify-the-cluster)
+    - [Troubleshooting](#troubleshooting)
+  - [Notes](#notes)
+
+
+
+## Prerequisites
+1. Ensure all machines (control plane and worker nodes) are running a compatible Linux OS.
+   
+## Steps
+1. Install Docker or another container runtime on each machine.
+2. Install `kubeadm`, `kubelet`, and `kubectl` on each machine.
+
+### Install Docker on Each Machine
+```sh
+    sudo apt-get update
+    sudo apt-get install -y apt-transport-https ca-certificates curl software-properties-common
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+    sudo apt-get update
+    sudo apt-get install -y docker-ce
+    sudo systemctl enable docker
+    sudo systemctl start docker
+```
+
+### Install Kubernetes Components on Each Machine
```

Comment:
I would specify this is only for debian based machines due to the use of apt. Granted I think anyone installing K8s should realize that if they are on a RHEL based system apt would not work.


I see the command `sudo apt-get install -y apt-transport-https ca-certificates curl` and `apt-get update` repeated. May be good to separate common commands out into their own sections. Something like `### Installation Prerequisites`

Reply:
Unless its not safe to repeat these commands(or others). I believe it is fine, and useful in the case some references just a portion of this document and adding more sections may just be cumbersome. I do see the value in some cases.